### ISSUE TITLE
ImageStacks does not write the space units fields in the NRRD header

### ISSUE THREAD

**Issue by muratmaga** [SlicerMorph/SlicerMorph#318]
This test prompted by [this user's question on Slicer forum](https://discourse.slicer.org/t/setting-up-units-in-slicer/34861/2)

1. Download the Bruker/Skyscan sample microCT dataset from SlicerMorph sample data
2. Import via ImageStacks with original resolution of 0.035mm (or 35 micron)s.
3. Export the datset and obtain the nrrd header. Note that there is no space units field to define it as millimeter.

I think we should support the space units in our own NRRD writer. 

@lassoan @pieper

**Comment by muratmaga** [comment]
Also if you set the units to Microns in Slicer, and then import the same dataset, this time entering 35 in space field, the imported data has the wrong scale. It is showing equivalent of it in meters:

<img width="876" alt="image" src="https://github.com/SlicerMorph/SlicerMorph/assets/21285140/9fb2e29e-c770-41c5-a6af-49c5b670918e">

**Comment by lassoan** [comment]
Using [`space units` field](https://teem.sourceforge.net/nrrd/format.html#spaceunits) is a good idea. We could then automatically convert spacing when reading an image.

If unit is not specified then probably the best is to assume that the unit is millimeters.

**Comment by muratmaga** [comment]
I assume this needs to be handled at Slicer level, not just Imagestacks. Do you want me to open an issue on the Slicer repo?

**Comment by lassoan** [comment]
Yes, to read/write units in nrrd files, the volume storage node in Slicer core would need to be updated. 

However, now that I think about it, we don't really need to write the unit into the header. We should just keep using the current convention consistently that by default the unit is millimeters.

**Comment by muratmaga** [comment]
But what about when the unit is set to micrometer in Slicer? Or cases where the nrrd file itself has the field space units sets to micron.

Ultimately this is a data integrity issues. Since unlike formats like ply or stl, nrrd indeed offers a mechanism to explicitly define the unit, we should use it.

Is the concern backward compatibility somehow?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
